app/src/main/java/com/android/unio/model/association/AssociationViewModel.kt

```diff
@@ -47,6 +50,24 @@ class AssociationViewModel(val repository: AssociationRepository) : ViewModel()
     }
   }
 
+  fun addAssociation(
+      inputStream: InputStream,
+      association: Association,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    viewModelScope.launch {
+      imageRepository.uploadImage(
+          inputStream,
+          "images/associations/${association.uid}",
```

Comment:
Hardcoded string. For URL, we should probably put them in a separate file (that contain only constant) to reuse them easily.